be/src/olap/bitmap_filter_predicate.h

```diff
@@ -0,0 +1,121 @@
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#pragma once
+
+#include "exprs/bitmapfilter_predicate.h"
+#include "exprs/runtime_filter.h"
+#include "olap/column_predicate.h"
+#include "vec/columns/column_dictionary.h"
+#include "vec/columns/column_nullable.h"
+#include "vec/columns/column_vector.h"
+#include "vec/columns/predicate_column.h"
+#include "vec/exprs/vruntimefilter_wrapper.h"
+
+namespace doris {
+
+// only use in runtime filter and segment v2
+template <PrimitiveType T>
+class BitmapFilterColumnPredicate : public ColumnPredicate {
+public:
+    using CppType = typename PrimitiveTypeTraits<T>::CppType;
+    using SpecificFilter = BitmapFilterFunc<T>;
+
+    BitmapFilterColumnPredicate(uint32_t column_id,
+                                const std::shared_ptr<BitmapFilterFuncBase>& filter, int)
+            : ColumnPredicate(column_id),
+              _filter(filter),
+              _specific_filter(static_cast<SpecificFilter*>(_filter.get())) {}
+    ~BitmapFilterColumnPredicate() override = default;
+
+    PredicateType type() const override { return PredicateType::BITMAP_FILTER; }
+
+    void evaluate(ColumnBlock* block, uint16_t* sel, uint16_t* size) const override;
+
+    void evaluate_or(ColumnBlock* block, uint16_t* sel, uint16_t size,
+                     bool* flags) const override {};
+    void evaluate_and(ColumnBlock* block, uint16_t* sel, uint16_t size,
+                      bool* flags) const override {};
+
+    Status evaluate(BitmapIndexIterator*, uint32_t, roaring::Roaring*) const override {
+        return Status::OK();
+    }
+
+    uint16_t evaluate(const vectorized::IColumn& column, uint16_t* sel,
+                      uint16_t size) const override;
+
+private:
+    template <bool is_nullable>
+    uint16_t evaluate(const vectorized::IColumn& column, const uint8_t* null_map, uint16_t* sel,
+                      uint16_t size) const {
+        if constexpr (is_nullable) {
+            DCHECK(null_map);
+        }
+
+        uint16_t new_size = 0;
+        new_size = _specific_filter->find_fixed_len_olap_engine(
+                (char*)reinterpret_cast<const vectorized::PredicateColumnType<T>*>(&column)
+                        ->get_data()
+                        .data(),
+                null_map, sel, size);
+        return new_size;
+    }
+
+    std::string _debug_string() override {
```

Comment:
it is better to make `debug_string` public,  support debug outside `BitmapFilterColumnPredicate`.

Reply:
There is a public `debug_string()` in the base class of `BitmapFilterColumnPredicate`, and `debug_string()` calls `_debug_string()`